### PR TITLE
BUILD-1970 Use Vault

### PR DESCRIPTION
**Pull request by tomverin**

### CODE REVIEW

**Comment by tomverin** [comment]
https://github.com/SonarSource/rspec/blob/master/.github/workflows/update_coverage.yml#L60 needs a specific githug token to be able to push to the repo

File: .github/workflows/create_new_rspec.yml [hunk 1]
```diff
@@ -12,6 +12,9 @@ on:
 jobs:
   create_new_rule:
     runs-on: ubuntu-latest
+    permissions:
+      contents: write
+      pull-requests: write
```
Comment (Godin) [inline .github/workflows/create_new_rspec.yml:17]: This change is unrelated to Vault.

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -8,6 +8,8 @@ on:
 jobs:
   build-and-deploy:
     runs-on: ubuntu-20.04
+    permissions:
+      contents: write
```
Comment (Godin) [inline .github/workflows/main.yml:12]: This change is unrelated to Vault.

File: .github/workflows/update_quickfix_status.yml [hunk 3]
```diff
@@ -26,6 +26,9 @@ jobs:
   update_quickfix_status:
     name: Update quick fix status
     runs-on: ubuntu-20.04
+    permissions:
+      contents: write
+      pull-requests: write
```
Comment (Godin) [inline .github/workflows/update_quickfix_status.yml:31]: This change is unrelated to Vault.

File: .cirrus.yml [hunk 4]
```diff
@@ -61,14 +66,14 @@ frontend_tests_task:
...
-      - cd frontend
-      - npm install
+    reupload_on_changes: "false" # since there is a fingerprint script
+  fingerprint_script:
+    - echo $CIRRUS_OS
+    - node --version
+    - cat frontend/package-lock.json
+  populate_script:
+    - cd frontend
+    - npm install
```
Comment (Godin) [inline .cirrus.yml:76]: This change is unrelated to Vault.

File: .cirrus.yml [hunk 5]
```diff
@@ -27,7 +32,7 @@ container_definition: &CONTAINER_DEFINITION
 
 setup_sonar_scanner: &SETUP_SONAR_SCANNER
   setup_sonar_scanner_script:
-    - apt update -y && apt upgrade -y && apt update -y && apt install -y unzip       
+    - apt update -y && apt upgrade -y && apt update -y && apt install -y unzip
```
Comment (Godin) [inline .cirrus.yml:35]: This change is unrelated to Vault.

**Review by Godin — CHANGES_REQUESTED** [review]

**Pull request merged**